File: start.py
import subprocess
import os
import sys
import time
import webbrowser
import platform
"python start.py"
# ANSI Colors
CYAN = '\033[96m'
GREEN = '\033[92m'
YELLOW = '\033[93m'
RED = '\033[91m'
BOLD = '\033[1m'
MAGENTA = '\033[95m'
NC = '\033[0m' # No Color

ASCII_ART = f"""
{CYAN}{BOLD}
   ____  ____  ________  __________  ____  ______   _____   _________________  ___
  / __ )/ __ \/  _/ / / / __ \/ __ \/ __ \/ __ \   / __ \ / ____/ ____/ __ \/_  /
 / __  / / / // // / / / / / / / / / / / / / / /  / /_/ // / __/ __/ / / / // / 
/ /_/ / /_/ // // /_/ / /_/ / /_/ / /_/ / /_/ /  / /_  // /_/ / /___/ /_/ // /  
/____/ \____/___/\____/_____/_____/\____/\____/  /_/ \_\\____/_____/_____//_/   
{NC}
{YELLOW}>> 建築法規 AI 權重分析系統 2.0 <<{NC}
"""

def type_text(text, delay=0.01):
    for char in text:
        sys.stdout.write(char)
        sys.stdout.flush()
        time.sleep(delay)
    print()

def kill_port(port):
    """Kills the process using the specified port."""
    try:
        if platform.system() == "Windows":
            result = subprocess.run(['netstat', '-ano'], capture_output=True, text=True)
            for line in result.stdout.splitlines():
                if f":{port}" in line and "LISTENING" in line:
                    parts = line.split()
                    pid = parts[-1]
                    type_text(f"{YELLOW}[!] 檢測到端口 {port} 被佔用，正在強制重啟伺服器... (PID: {pid}){NC}")
                    subprocess.run(['taskkill', '/F', '/PID', pid], capture_output=True)
                    time.sleep(1)
                    return True
    except Exception:
        pass
    return False

def main():
    os.system('cls' if os.name == 'nt' else 'clear')
    print(ASCII_ART)
    
    type_text(f"{MAGENTA}正在初始化核心系統模組...{NC}", 0.02)
    time.sleep(0.5)
    
    kill_port(9000)
    
    app_dir = os.path.join(os.getcwd(), 'app')
    if not os.path.exists(os.path.join(app_dir, 'package.json')):
        print(f"{RED}[X] 錯誤: 找不到 app 目錄或 package.json{NC}")
        return

    print(f"\n{CYAN}--- 啟動流程 ---{NC}")
    
    # Check node_modules
    if not os.path.exists(os.path.join(app_dir, 'node_modules')):
        print(f"{YELLOW}[1/3] 正在安裝必要元件 (初次運行需較長時間)...{NC}")
        subprocess.run(['npm', 'install'], cwd=app_dir, shell=True)
    else:
        print(f"{GREEN}[1/3] 元件庫檢查通過{NC}")

    print(f"{YELLOW}[2/3] 正在喚醒 Vite 伺服器...{NC}")
    
    # Run dev server
    cmd = "cmd /c npm run dev"
    try:
        process = subprocess.Popen(cmd, cwd=app_dir, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, text=True)
        
        # Open browser after a small delay
        time.sleep(3)
        print(f"{GREEN}[3/3] 正在串接瀏覽器介面...{NC}")
        webbrowser.open("http://127.0.0.1:9000")
        
        print(f"\n{BOLD}{GREEN}==========================================={NC}")
        print(f"{BOLD}{GREEN}   ⚡ 系統掛載完畢，可開始使用！{NC}")
        print(f"{BOLD}{GREEN}   URL: http://127.0.0.1:9000{NC}")
        print(f"{BOLD}{GREEN}==========================================={NC}\n")
        
        print(f"{CYAN}伺服器日誌監控中 (Ctrl+C 結束行程):{NC}")
        
        # Stream the output
        for line in process.stdout:
            if "VITE" in line or "ready" in line:
                print(f"{MAGENTA}> {line.strip()}{NC}")
            elif "Error" in line:
                print(f"{RED}> {line.strip()}{NC}")
            
    except KeyboardInterrupt:
        print(f"\n{YELLOW}[!] 正在執行關機程序...{NC}")
        process.terminate()
        print(f"{GREEN}[V] 系統已安全中斷。{NC}")
    except Exception as e:
        print(f"{RED}[X] 嚴重錯誤: {e}{NC}")

if __name__ == "__main__":
    main()
".\啟動系統.bat"

File: start.js
const { spawn, execSync } = require('child_process');
const path = require('path');
const fs = require('fs');
const os = require('os');

// ANSI Colors
const CYAN = '\x1b[96m';
const GREEN = '\x1b[92m';
const YELLOW = '\x1b[93m';
const RED = '\x1b[91m';
const BOLD = '\x1b[1m';
const MAGENTA = '\x1b[95m';
const NC = '\x1b[0m'; // No Color

const ASCII_ART = `
${CYAN}${BOLD}
   ____  ____  ________  __________  ____  ______   _____   _________________  ___
  / __ )/ __ \\/  _/ / / / __ \\/ __ \\/ __ \\/ __ \\   / __ \\ / ____/ ____/ __ \\/_  /
 / __  / / / // // / / / / / / / / / / / / / / /  / /_/ // / __/ __/ / / / // / 
/ /_/ / /_/ // // /_/ / /_/ / /_/ / /_/ / /_/ /  / /_  // /_/ / /___/ /_/ // /  
/____/ \\____/___/\\____/_____/_____/\\____/\\____/  /_/ \\_\\\\____/_____/_____//_/   
${NC}
${YELLOW}>> 建築法規 AI 權重分析系統 3.0 (Node Edition) <<${NC}
`;

function typeText(text, delay = 10) {
    return new Promise(resolve => {
        let i = 0;
        const interval = setInterval(() => {
            process.stdout.write(text[i]);
            i++;
            if (i >= text.length) {
                clearInterval(interval);
                process.stdout.write('\n');
                resolve();
            }
        }, delay);
    });
}

function killPort(port) {
    try {
        if (os.platform() === 'win32') {
            const output = execSync(`netstat -ano | findstr :${port}`).toString();
            const lines = output.split('\n');
            for (const line of lines) {
                if (line.includes(`:${port}`) && line.includes('LISTENING')) {
                    const parts = line.trim().split(/\s+/);
                    const pid = parts[parts.length - 1];
                    console.log(`${YELLOW}[!] 正在清理佔用端口 ${port} 的進程 (PID: ${pid})...${NC}`);
                    execSync(`taskkill /F /PID ${pid}`, { stdio: 'ignore' });
                    return true;
                }
            }
        }
    } catch (e) {
        // Silently fail if no process found
    }
    return false;
}

async function main() {
    process.stdout.write('\x1Bc'); // Clear screen
    console.log(ASCII_ART);
    
    await typeText(`${MAGENTA}正在初始化核心系統模組...${NC}`, 10);
    
    killPort(9000);
    
    const appDir = path.join(process.cwd(), 'app');
    if (!fs.existsSync(path.join(appDir, 'package.json'))) {
        console.log(`${RED}[X] 錯誤: 找不到 app 目錄或 package.json${NC}`);
        return;
    }

    console.log(`\n${CYAN}--- 啟動流程 ---${NC}`);
    
    if (!fs.existsSync(path.join(appDir, 'node_modules'))) {
        console.log(`${YELLOW}[1/3] 正在安裝必要元件 (初次運行需較長時間)...${NC}`);
        execSync('npm install', { cwd: appDir, stdio: 'inherit' });
    } else {
        console.log(`${GREEN}[1/3] 元件庫檢查通過${NC}`);
    }

    console.log(`${YELLOW}[2/3] 正在喚醒 Vite 伺服器...${NC}`);
    
    const npm = os.platform() === 'win32' ? 'npm.cmd' : 'npm';
    const server = spawn(npm, ['run', 'dev'], { 
        cwd: appDir,
        shell: true,
        stdio: ['inherit', 'pipe', 'pipe']
    });

    server.stdout.on('data', (data) => {
        const line = data.toString();
        if (line.includes('VITE') || line.includes('ready')) {
            process.stdout.write(`${MAGENTA}> ${line.trim()}${NC}\n`);
        }
    });

    setTimeout(() => {
        console.log(`${GREEN}[3/3] 正在串接瀏覽器介面...${NC}`);
        const url = 'http://127.0.0.1:9000';
        const start = (process.platform === 'darwin' ? 'open' : process.platform === 'win32' ? 'start' : 'xdg-open');
        require('child_process').exec(`${start} ${url}`);
        
        console.log(`\n${BOLD}${GREEN}===========================================${NC}`);
        console.log(`${BOLD}${GREEN}   ⚡ 系統掛載完畢，可開始使用！${NC}`);
        console.log(`${BOLD}${GREEN}   URL: ${url}${NC}`);
        console.log(`${BOLD}${GREEN}===========================================${NC}\n`);
        console.log(`${CYAN}提示: 按下 Ctrl+C 可以關閉系統。${NC}`);
    }, 4000);

    process.on('SIGINT', () => {
        console.log(`\n${YELLOW}[!] 正在執行關機程序...${NC}`);
        server.kill();
        process.exit();
    });
}

main();
